tools/StaticAnalysis/Exceptions/AzureRM.Compute/BreakingChangeIssues.csv

```diff
@@ -21,3 +21,8 @@
 "Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.Automation.UpdateAzureRmVmssVM","Update-AzureRmVmssVM","0","3000","The type of property 'CreateOption' of type 'Microsoft.Azure.Commands.Compute.Models.PSVirtualMachineDataDisk' has changed from 'System.String' to 'Microsoft.Azure.Management.Compute.Models.DiskCreateOptionTypes'.","Change the type of property 'CreateOption' back to 'System.String'."
 "Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.Automation.NewAzureRmVmssIpConfigCommand","New-AzureRmVmssIpConfig","0","3000","The type of property 'PrivateIPAddressVersion' of type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSetIPConfiguration' has changed from 'System.String' to 'System.Nullable`1[Microsoft.Azure.Management.Compute.Models.IPVersion]'.","Change the type of property 'PrivateIPAddressVersion' back to 'System.String'."
 "Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.Automation.RemoveAzureRmContainerService","Remove-AzureRmContainerService","0","1020","The cmdlet 'Remove-AzureRmContainerService' no longer has output type 'System.Void'.","Make cmdlet 'Remove-AzureRmContainerService' return type 'System.Void'."
+"Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.RemoveAzureRmVmssDiagnosticsExtension","Remove-AzureRmVmssDiagnosticsExtension","0","3010","The property 'IdentityIds' of type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSetIdentity' has been removed.","Add the property 'IdentityIds' back to type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSetIdentity'."
+"Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.NewAzureAdditionalUnattendContentCommand","Add-AzureRmVMAdditionalUnattendContent","0","3010","The property 'IdentityIds' of type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineIdentity' has been removed.","Add the property 'IdentityIds' back to type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineIdentity'."
```

Comment:
Removing `IdentityIds` is a breaking change. You aren't supposed to expose your SDK models. For this PR to be accepted, you need a model that matches the current set of properties in the exposed type. Otherwise, scripts using this property will be broken.

Reply:
@MiYanni These types are from client library.  Should I create PS type (say PSVirtualMachineIdentity) for each of these types, and change the return type from VirtualMachineIdentity to PSVirtualMachineIdentity ?  (Isn't this another breaking change?  If not, I will create PS types and do mapping.

Reply:
Yes, that is how it should be handled. `PSVirtualMachineIdentity` must have the same properties as what the prior `VirtualMachineIdentity` had. The analyzer checks public properties of that type, and then public properties of the types within those properties, and so-forth, in a tree to make sure they are the same from one version to the next version. So, as long as that is the same, you shouldn't get a failure, and should be able to remove these exceptions.